### ISSUE TITLE
Add vault outline / explorer to the sidebar

### ISSUE THREAD

**Issue by KosmosisDire** [KosmosisDire/obsidian-webpage-export#30]
Add a vault outline that would show all the files and file structure in an exported vault or folder. This would be inserted into the left sidebar.

**Comment by c-jw-Medic** [comment]
I came here to suggest this exact feature.  So instead, I'll just give a big +1 to the vault outline feature.

**Comment by KosmosisDire** [comment]
This is definitely something I plan to do, but it's a surprisingly difficult thing to implement since I allow people to export arbitrary sets of files and folders, not only the whole vault. However I definitely plan to add this in the future

**Comment by Bagel03** [comment]
As far as I can tell, all the needed styles are already included, so could it be possible to just expand all closed tabs in the sidebar and export `$(".nav-files-container")` (with minimal changes) when exporting a whole vault?

**Comment by KosmosisDire** [comment]
I will likely use the included styles, however as I stated in my above comment the difficulty is that I need to be able to generate a list that only includes the specific files the user is exporting. Additionally the nav may be collapsed. The built-in nav also is controlled by a script not by links. So I would also have to write a script to control it or modify it. So in the end I think it is better and cleaner to just generate one from scratch.

**Comment by KosmosisDire** [comment]
Update on this issue, as I am currently implementing it:
![image](https://user-images.githubusercontent.com/39423700/236427989-1430f86a-c378-43f3-ac57-244f6bf6193c.png)

You can see the vault outline in the left sidebar along with the name of the vault above it. (This is a test vault)

If you want you  can access a live demo here, although it may be broken or in develpment when you go there since I am actively using this link for testing: https://nathang.dev/fileshare/misc/graph_view_demo/test/ultimate-test-document.html

It is coming soon!

**Comment by berthelemy** [comment]
This is looking great! Thank you.

**Comment by aubreyz** [comment]
I AGREE THIS IS LOOKING GREAT. 
One alternative possibility which might put power in user's hands and open up more possibilities than a folder structure is to allow user to have a single note tagged in some way - maybe with YAML, maybe in the settings) where that note goes into the Left sidebar. That way the user can put whatever they like there, like key links, a home link or whatever.

**Comment by KosmosisDire** [comment]
> One alternative possibility which might put power in user's hands and open up more possibilities than a folder structure is to allow user to have a single note tagged in some way - maybe with YAML, maybe in the settings) where that note goes into the Left sidebar. That way the user can put whatever they like there, like key links, a home link or whatever. 

I want to do something like this in the future, which is to let the user drag around the different elements in a little editor before exporting. It would also allow the user to add their own HTML or markdown as elements they could add to the exported interface.

This is a ways away, but it would allow what you just said as well as almost anything else the user could think of.

**Comment by aubreyz** [comment]
> I want to do something like this in the future, which is to let the user drag around the different elements in a little editor before exporting. It would also allow the user to add their own HTML or markdown as elements they could add to the exported interface. As well as change the position of those elements (and the built in ones) in the interface.

Big things start with a tiny step - Can I make a small suggestion as to what that tiny step should be:
I haven't made time to study the structure of the HTML file output, but it would be nice to have an token in a comment in the HTML which is where some user code is to be inserted. 

Probably either where the graph is or just above that, or on the right under the day/night toggle. 

Users who are interested will be able to search and replace that token (with grep or whatever) with any code they want. That would seem to be a good way to launch the path of enabling creation of proper websites.

**Comment by KosmosisDire** [comment]
> @aubreyz  but it would be nice to have an token in a comment in the HTML which is where some user code is to be inserted.

Thanks for the suggestion, I probably won't do this simply because it is a step which doesn't actually lead towards the final goal. In other words, if I add that feature it would become obsolete as soon as I implement the features I described above. This would then mean I would have to either remove the feature which was already part a few people's workflow, or have to keep supporting a feature which is completely obsolete (in this case is is a very small feature, but one which would exist in code and I would have to keep it updated as the code changes)

The other reason I'd rather not do that is because it is a feature that is completely undiscoverable. The only way a user could discover it would be to read some piece of documentation about it and I really want to avoid that at all costs. If I did add it into the UI for discoverability, that would increase the burden of supporting the feature once it is obsolete.

However, the features that I described can have tiny steps as well so I will likely start there when I do decide to start implementing those features.

I guess it is mainly about time; you want to start using this feature soon. I do understand, but I probably will take a while with the feature because I would rather put out something polished in a lot of time than quick and dirty that might be regretted later.

But thank you for the suggestion! It is really really helpful for people to give their ideas and suggestions, and I would be happy to hear your thoughts on this :)
